analyzer/consensus/consensus.go

```diff
@@ -300,6 +284,36 @@ func (m *processor) ProcessBlock(ctx context.Context, uheight uint64) error {
 		return err
 	}
 
+	return nil
+}
+
+// Implements BlockProcessor interface.
+func (m *processor) ProcessBlock(ctx context.Context, uheight uint64) error {
+	if uheight > math.MaxInt64 {
+		return fmt.Errorf("height %d is too large", uheight)
+	}
+	height := int64(uheight)
+
+	// Fetch all data.
+	fetchTimer := m.metrics.BlockFetchLatencies()
+	data, err := fetchAllData(ctx, m.source, m.network, height, m.mode == analyzer.FastSyncMode)
+	if err != nil {
+		if strings.Contains(err.Error(), fmt.Sprintf("%d must be less than or equal to the current blockchain height", height)) {
+			return analyzer.ErrOutOfRange
+		}
+		return err
+	}
+	fetchTimer.ObserveDuration() // We make no observation in case of a data fetch error; those timings are misleading.
+
+	// Process data, prepare updates.
+	analysisTimer := m.metrics.BlockAnalysisLatencies()
+	batch := &storage.QueryBatch{}
+	err = m.queueDbUpdates(batch, *data)
+	analysisTimer.ObserveDuration()
+	if err != nil {
+		return err
+	}
```

Comment:
Hm what do you think of only observing when the analysis succeeds?

Reply:
No very strong opinion, but I'm leaning towards marking it always. Just like we do for db txs, for example; we time it even if they need to be rolled back because of a deadlock. I think that makes sense; we're still spending time on analysis, even if it eventually fails, and the goal of the metrics is to help us understand where time is being spent.